libqtile/backend/x11/window.py

```diff
@@ -892,21 +886,232 @@ def place(
+
+        # sort the flags from bottom to top, True meaning further below than False at each step
+        states = [
+            desktop,
+            below,
+            above or (dock and not below),
+            full and focus,
+        ]
+        other = not any(states)
+        states.insert(2, other)
+
+        # This next bit could have been done by adding more conditions when setting the
+        # states flags but it's easier to look for the highest matching level and then
+        # set all other levels to False
+        highest = max(i for i, state in enumerate(states) if state)
+        win_states = [True if i == highest else False for i in range(5)]
+
+        return (win_states[0], win_states[1], win_states[2], win_states[3], win_states[4])
+
+    def change_layer(self, up=True, top_bottom=False, kwargs=None):
+        """Raise a window above its peers or move it below them, depending on 'up'.
+        Raising a normal window will not lift it above pinned windows etc.
+
+        There are a few important things to take note of when relaying windows:
+        1. If a window has a defined parent, it should not be moved underneath it.
+           In case children are blocking, this could leave an application in an unusable state.
+        2. If a window has children, they should be moved along with it.
+        3. If a window has a defined parent, either move the parent or do nothing at all.
+        4. EMWH-flags follow strict layering rules:
+           https://specifications.freedesktop.org/wm-spec/1.3/ar01s07.html#STACKINGORDER
+        """
+        if len(self.qtile.windows_map) < 2:
+            return
+
+        parent = self.window.get_wm_transient_for()
+        if parent is not None and not up:
+            if kwargs:
+                self.window.configure(**kwargs)
+            return
+
+        layering = self.get_layering_information()
+
+        # Comparison of layer states: -1 if window is now in a lower state group,
+        # 0 if it's in the same group and 1 if it's in a higher group
+        moved = (self.previous_layer > layering) - (layering > self.previous_layer)
+        self.previous_layer = layering
+
+        stack = list(self.qtile.core._root.query_tree())
+        if self.wid not in stack or len(stack) < 2:
+            if kwargs:
+                self.window.configure(**kwargs)
+            return
+
+        # Get list of windows that are in the stack and managed by qtile
+        # List of tuples (XWindow object, transient_for, layering_information)
+        windows = list(
+            map(
+                lambda w: (
+                    w.window,
+                    w.window.get_wm_transient_for(),
+                    w.get_layering_information(),
+                ),
+                [
+                    self.qtile.windows_map[i]
+                    for i in stack
+                    if i in self.qtile.windows_map
+                    and isinstance(self.qtile.windows_map[i], _Window)
```

Comment:
Aren't all values in `self.qtile.windows_map` going to be instances of `_Window`?

Reply:
That should be right if we've stopped `Systray` icons from being included in the map. They're `XWindow`s which don't inherit from `_Window`.